### PR TITLE
feat(TKC-1580): prepare Init Process for TestWorkflow containers

### PR DESCRIPTION
**Pull request by rangoo94**
## Pull request description 

Prepare Init Process that will wrap all TestWorkflows operations in the containers. Additionally, it includes some fixes and improvements to the Expressions Engine (force simplifying unknown structs, better boolean prediction).

It supports:
* Skipping container execution
* Passing status to parents/children
* Sharing state across different containers
* Aborting steps
* Negative expectation
* Timeouts (nested already)
* Retry (single step only)
* Computing variables locally, i.e. reading files (`file("/data/some/path")`), reading shared data

In the future, we need to support:
* Nested retries (sharing iterations is already there)
* Obfuscating sensitive data in the output
* Saving the data from the output clauses in the shared state

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: cmd/tcl/testworkflow-init/main.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+		if err != nil {
+			fmt.Printf(`resolving command: %s: %s\n`, shellquote.Join(original...), err.Error())
+			os.Exit(155)
+		}
+	}
+
+	// Fail when there is nothing to run
+	if len(args) == 0 {
+		fmt.Println("missing command to run")
+		os.Exit(189)
```
Comment (vsukhin) [inline cmd/tcl/testworkflow-init/main.go:172]: I would use const

File: cmd/tcl/testworkflow-init/data/state.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+	}
+	str, _ := v.Static().StringValue()
+	return str
+}
+
+func readState(filePath string) {
+	b, err := os.ReadFile(filePath)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			panic(err.Error())
```
Comment (vsukhin) [inline cmd/tcl/testworkflow-init/data/state.go:84]: do we need panic everywhere?

**Review by vsukhin — APPROVED** [review]

File: cmd/tcl/testworkflow-init/data/state.go (hunk 2 above)
Comment (rangoo94) [reply cmd/tcl/testworkflow-init/data/state.go:84]: Basically, these are clearly not User issues, but likely actual implementation bugs if happens, nothing that User may deal with

Comment (rangoo94) [reply cmd/tcl/testworkflow-init/data/state.go]: Added `Failf` helper, and used in in places where error code is expected; `panic` is left in:

* places which could be used by other programs (i.e. emit)
* places where something unusual happened, likely due implementation bug (i.e. state not saved), not user error

**Pull request merged**